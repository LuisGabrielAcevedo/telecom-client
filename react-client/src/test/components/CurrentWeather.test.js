import React from "react";
import "@testing-library/jest-dom";
import { shallow, mount } from "enzyme";
import CurrentWeather from "../../components/CurrentWeatherComponent";
import { act } from "react-dom/test-utils";

describe("CurrentWeather Component tests", () => {
  const mockGeolocation = {
    getCurrentPosition: jest.fn().mockImplementationOnce((success) =>
      Promise.resolve(
        success({
          coords: {
            latitude: 51.1,
            longitude: 45.3,
          },
        })
      )
    ),
  };
  global.navigator.geolocation = mockGeolocation;

  it("should init the component", () => {
    const wrapper = shallow(<CurrentWeather />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should distpatch the hooks and update the component", async () => {
    const wrapper = mount(<CurrentWeather />);
    await act(async () => wrapper);
    wrapper.update();
  });
});
