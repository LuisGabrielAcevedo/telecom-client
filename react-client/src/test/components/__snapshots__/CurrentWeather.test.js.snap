// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrentWeather Component tests should init the component 1`] = `
<div>
  <h1>
    Clima actual:
  </h1>
  <Loading
    repeat={1}
  />
</div>
`;
